src/ui/Templates/TemplateToHtml.ts

```diff
@@ -0,0 +1,72 @@
+import { IQueryResults } from '../../rest/QueryResults';
+import { IQueryResult } from '../../rest/QueryResult';
+import { Assert, QueryUtils, Component, Initialization, QueryStateModel } from '../../Core';
+import { IInitResult } from '../Base/Initialization';
+import { Template } from './Template';
+import { SearchInterface } from '../../ui/SearchInterface/SearchInterface';
+import { ResultList } from '../ResultList/ResultList';
+import { ValidLayout } from '../ResultLayoutSelector/ValidLayout';
+import { pluck, sortBy, map } from 'underscore';
+
+export interface ITemplateToHtml {
+  resultTemplate: Template;
+  queryStateModel: QueryStateModel;
+  searchInterface: SearchInterface;
+}
+
+export class TemplateToHtml {
```

Comment:
I don't think we should expose this class (and the ITemplateToHtml) interface in the public reference documentation; seems rather internal to me.

So unless you think otherwise, that's pretty much going to be my review documentation-wise :)